blockbuilder/genesisState.go

```diff
@@ -0,0 +1,33 @@
+package blockbuilder
+
+import (
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/validator"
+)
+
+type GenesisConfig struct {
+	ChainID                 string           `json:"chain_id" validate:"required"`
+	WhitelistedSequencerSet []felt.Felt      `json:"whitelisted_sequencer_set"`
+	State                   genesisStateData `json:"state"`
+}
+
+type genesisStateData struct {
+	Classes       []string                          `json:"classes"` // []path-to-class.json
+	Contracts     map[felt.Felt]genesisContractData `json:"contracts"`
+	FunctionCalls []functionCall                    `json:"function_calls"`
+}
+
+type genesisContractData struct {
+	ClassHash       felt.Felt   `json:"class_hash"`
+	ConstructorArgs []felt.Felt `json:"constructor_args"`
+}
```

Comment:
I think one of y'all suggested that, but I'm not sure we ever made a final decision about it.

Reply:
`deployer_address` is tricky to set in the genesis context tho, no?

Reply:
We could just print the contract addresses we calculate at startup? Eg, maybe we can just set `deployer_address` to zero, and salt to zero / some random number?

Reply:
If we allow them to supply the `salt` value, we can compute a unique address for them, and just print it on startup. This means we can redefine/simplify
`Contracts map[felt.Felt]genesisContractData` 
to 
`Contracts {classHash, constructorArgs, Salt}`

Reply:
I guess we can also allow them to specify `deployer_address`, afaik, we don't actually need this address to be initialised. It's just another argument into the address calculation.

Reply:
> afaik, we don't actually need this address to be initialised.
Yes, but still it is meant to be in the normal execution. So I am afraid it might come back to bite us at some point.

I don't see a downside of having using hardcode the initial contract addresses.

Reply:
Okay, so because the user will have to know the contract address anyway (to use `FunctionCalls` - assuming they want to use them) we might as well make them supply it upfront. In that case I'll leave this as 
`Contracts     map[string]GenesisContractData `

```
type GenesisContractData struct {
	ClassHash       felt.Felt   `json:"class_hash"`
	ConstructorArgs []felt.Felt `json:"constructor_args"`
}
```